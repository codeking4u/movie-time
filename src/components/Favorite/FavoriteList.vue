<template>
  <div class="favorite-container section">
    <h2>Favourites</h2>
    <template v-if="allFavorites.length">
      <div class="favorite-list flex">
        <FavoriteShow
          v-for="fav in allFavorites"
          :key="fav.id"
          :favorite="fav"
          @delete="removeFavorite"
        />
      </div>
    </template>
    <template v-else>
      <p>No favourites added yet. Click on <i class="material-icons">favorite</i> icon to add.</p>
    </template>
  </div>
</template>

<script lang="ts">
import { computed } from 'vue'
import { useStore } from 'vuex'
import FavoriteShow from './FavoriteShow.vue'

export default {
  components: { FavoriteShow },
  setup() {
    const store = useStore()
    const allFavorites = computed(() => store.getters.getAllFavorites)

    const removeFavorite = (index: Number) => {
      console.log('delete')
      store.commit('removeFromFavorites', index)
    }

    return {
      allFavorites,
      removeFavorite
    }
  }
}
</script>

<style lang="scss" scoped>
.favorite-list {
  display: grid;
  grid-template-columns: repeat(auto-fit, minmax(250px, 1fr));
  align-items: center;
  grid-gap: 25px;
}
</style>
