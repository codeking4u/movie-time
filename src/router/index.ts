import { createRouter, createWebHistory } from 'vue-router'
import HomePage from '../views/HomePage.vue'
import SearchResult from '../views/SearchResult.vue'
import ShowDetail from '../views/ShowDetail.vue'

const routes = [
  {
    path: '/',
    name: 'HomePage',
    component: HomePage,
    meta: { title: 'Dashboard' }
  },
  {
    path: '/search',
    name: 'SearchResult',
    component: SearchResult,
    meta: { title: 'Search' }
  },
  {
    path: '/show/:id',
    name: 'ShowDetail',
    component: ShowDetail,
    props: true,
    meta: { title: 'Show' }
  }
]
const router = createRouter({
  history: createWebHistory(import.meta.env.BASE_URL),
  routes: routes
})

export default router
