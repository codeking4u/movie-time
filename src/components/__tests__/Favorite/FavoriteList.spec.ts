import { describe, it, expect } from 'vitest'
import { mount } from '@vue/test-utils'
import { createStore } from 'vuex'
import FavoriteList from '@/components/Favorite/FavoriteList.vue'
import router from '@/router'

describe('FavoriteList.vue', () => {
  it('renders correctly', () => {
    const mockStore = createStore({
      getters: {
        getAllFavorites: () => [{ id: 1, name: 'Favorite 1' }]
      }
    })

    const wrapper = mount(FavoriteList, {
      global: {
        plugins: [mockStore, router]
      }
    })

    expect(wrapper.html()).toMatchSnapshot()
  })
})
