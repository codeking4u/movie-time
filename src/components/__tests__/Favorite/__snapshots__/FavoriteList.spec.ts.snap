// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`FavoriteList.vue > renders correctly 1`] = `
"<div data-v-1eff25f7=\\"\\" class=\\"favorite-container section\\">
  <h2 data-v-1eff25f7=\\"\\">Favourites</h2>
  <div data-v-1eff25f7=\\"\\" class=\\"favorite-list flex\\"><a data-v-73416b1a=\\"\\" data-v-1eff25f7=\\"\\" href=\\"/show/1\\" class=\\"favorite flex flex-ai-c\\">
      <div data-v-73416b1a=\\"\\" class=\\"favorite__img\\"><img data-v-73416b1a=\\"\\" src=\\"/src/assets/images/show-no-image.png\\" alt=\\"Show image\\"></div>
      <div data-v-73416b1a=\\"\\" class=\\"favDetails flex flex-jc-sb\\">
        <p data-v-73416b1a=\\"\\" class=\\"favDetails__name\\">Favorite 1</p>
        <div data-v-73416b1a=\\"\\" class=\\"favDetails__delete flex flex-jc-c flex-ai-c\\"><button data-v-73416b1a=\\"\\" class=\\"delete-icon\\"><i data-v-73416b1a=\\"\\" class=\\"material-icons\\">clear</i></button></div>
      </div>
    </a></div>
</div>"
`;
